frame/support/src/inherent.rs

```diff
@@ -135,14 +168,53 @@ macro_rules! impl_outer_inherent {
 				result
 			}
 		}
+
+		impl $crate::traits::InherentPositionCheck<$block> for $runtime {
+			fn check_inherent_position(block: &$block) -> Result<(), u32> {
+				use $crate::inherent::ProvideInherent;
+				use $crate::traits::{IsSubType, ExtrinsicCall};
+				use $crate::sp_runtime::traits::Block as _;
+
+				let mut checking_for_inherents = true;
+
+				for (i, xt) in block.extrinsics().iter().enumerate() {
+					let is_signed = $crate::inherent::Extrinsic::is_signed(xt).unwrap_or(false);
+
+					let is_inherent = if is_signed {
+						// Signed extrinsics are not inherents.
+						false
+					} else {
+						let mut is_inherent = false;
```

Comment:
Is it correct to say if this never gets changed in any of the following repetitions, then it is unsigned?

Reply:
yes, an unsigned extrinsics is considered an inherent if a pallet say it is one, otherwise it is an unsigned transaction (not-inherent).